posts/2024-06-04-24.0.0.6-beta.adoc

```diff
@@ -0,0 +1,455 @@
+blog_description: The 24.0.0.6-beta release includes previews of the Jakarta Validation and Jakarta Data implementations, both of which are part of Jakarta EE 11. The release also includes improvements for MicroProfile Metrics in MicroProfile 3.0 and 4.0 and InstantOn support for distributed HTTP session caching.
+open-graph-image: https://openliberty.io/img/twitter_card.jpg
+open-graph-image-alt: Open Liberty Logo
+---
+= Jakarta EE11 previews and more in 24.0.0.6-beta
+David Mueller <https://github.com/dmuelle>
+:imagesdir: /
+:url-prefix:
+:url-about: /
+//Blank line here is necessary before starting the body of the post.
+
+
+The 24.0.0.6-beta release includes previews of the Jakarta Validation and Jakarta Data implementations, both of which are part of Jakarta EE 11. The release also includes enhancements for histogram and timer metrics in MicroProfile 3.0 and 4.0 and InstantOn support for distributed HTTP session caching.
+
+The link:{url-about}[Open Liberty] 24.0.0.6-beta includes the following beta features (along with link:{url-prefix}/docs/latest/reference/feature/feature-overview.html[all GA features]):
+
+* <<validate, Validate Java constraints and records with Jakarta Validation 3.1>>
+* <<data, Preview of Jakarta Data (Release Candidate 1)>>
+* <<mpm, Enhanced histogram and timer metrics in MicroProfile 3.0 and 4.0>>
+* <<jcache, InstantOn support distributed HTTP session caching>>
+
+
+// // // // // // // //
+// In the preceding section:
+// Change SUB_FEATURE_TITLE to the feature that is included in this release and
+// change the SUB_TAG_1/2/3 to the heading tags
+//
+// However if there's only 1 new feature, delete the previous section and change it to the following sentence:
+// "The link:{url-about}[Open Liberty] 24.0.0.6-beta includes SUB_FEATURE_TITLE"
+// // // // // // // //
+
+See also link:{url-prefix}/blog/?search=beta&key=tag[previous Open Liberty beta blog posts].
+
+// // // // DO NOT MODIFY THIS COMMENT BLOCK <GHA-BLOG-TOPIC> // // // //
+// Blog issue: https://github.com/OpenLiberty/open-liberty/issues/28397
+// Contact/Reviewer: mswatosh
+// // // // // // // //
+[#validate]
+== Validate Java constraints and records with Jakarta Validation 3.1
+
+Jakarta Validation provides a mechanism to validate constraints that are imposed on Java objects by using annotations. The most noticeable change in version 3.1 is the name change, from Jakarta Bean Validation to  Jakarta Validation. This version also includes explicit support for validating Java record classes, which were added in Java 16. Records reduce much the boilerplate code in simple data classes, and pair nicely with Jakarta Validation.
+
+<_insert explanation of these code examples_>

```

Comment:
@mswatosh can you provide a brief description of the context and value of these examples?

Reply:
```suggestion
The following example demonstrates the simplicity of defining a Java record with Jakarta Validation annotations, as well as performing validation.```